lib/cylc/cylc_xdot.py

```diff
@@ -417,14 +413,18 @@ def get_graph(self, group_nodes=None, ungroup_nodes=None,
         family_nodes = self.suiterc.get_first_parent_descendants()
         # Note this is used by "cylc graph" but not gcylc.
         # self.start_ and self.stop_point_string come from CLI.
+
+        tbg_color = getattr(self.style, 'bg', None)[gtk.STATE_NORMAL]
+        tfg_color = getattr(self.style, 'fg', None)[gtk.STATE_NORMAL]
```

Comment:
I've found that GTK sometimes returns colours using one character per channel rather than the conventional two (i.e. `#000` rather than `#000000`) which causes "Unknown color" warning messages to appear and the color not to be set.

Simple fix:

```python
'#' + ''.join('%02x' % (x * 255) for x in (c.red_float, c.green_float, c.blue_float))
```

Reply:
Yes, I've actually done this (in a less elegant way) where the GTK colors are used. I'll flag it...

Reply:
@sadielbartholomew This *should* fix the `#fff` warning and invisible graph elements problems you reported.